### ISSUE TITLE
[import/no-internal-modules] Type can be imported from an internal path without an error

### ISSUE THREAD

**Issue by lkappeler** [benmosher/eslint-plugin-import#1880]
The no internal modules rule does not lint when importing a typescript type from an internal path.

```typescript
// Error from the linter
import { myFunction } from "@namespace/package/deep/nested/path/index"
// No error from the linter
import { MyType } from "@namespace/package/deep/nested/path/index"
```

**Comment by JounQin** [comment]
Typings do not affect the final output, so I would say it is as expected, or maybe a new option can be added, but I don't think it's worth.

And if you use https://github.com/import-js/eslint-import-resolver-typescript with `exports` field in `package.json`, it should work as expected.
